splunklib/searchcommands/search_command.py

```diff
@@ -173,6 +173,14 @@ def logging_level(self, value):
                 raise ValueError('Unrecognized logging level: {}'.format(value))
         self._logger.setLevel(level)
 
+    def add_field(self, current_record, field_name, field_value):
+        self._record_writer.custom_fields.add(field_name)
+        current_record[field_name] = field_value
```

Comment:
This function should return current_record or else the documentation is incorrect.
```suggestion
    def add_field(self, current_record, field_name, field_value):
        self._record_writer.custom_fields.add(field_name)
        current_record[field_name] = field_value
        return current_record
```

Reply:
Good catch @Bre77 - I think in this case the documentation needs to be updated, the `current_record` should be updated with the new field without the need to return since it's essentially pass-by-reference for this usage (I always go back to this article when I need to refresh my memory: https://robertheaton.com/2014/02/09/pythons-pass-by-object-reference-as-explained-by-philip-k-dick/).

So if you all agree can we fix the README.md above to the following?
```
class CustomStreamingCommand(StreamingCommand):
    def stream(self, records):
        for index, record in enumerate(records):
            if index % 1 == 0:
                self.add_field(record, "odd_record", "true")
            yield record
```

Reply:
And to make sure that example we provide is working ^ that's probably a good implementation to add when adding tests for the StreamingCommand / add_field case

Reply:
I've changed my implementation to using pass by reference (as per the suggested doco update) and everything works as expected.

Reply:
Good catch @Bre77 
I've updated the README accordingly.